docs/autoencoders.md

```diff
@@ -0,0 +1,88 @@
+# Autoencoders
+
+An Autoencoder is a type of Artificial Neural Network, used to learn efficient data representations (encoding) of unlabeled data. 
+
+It mainly consist of 2 components: an encoder and a decoder. The encoder compresses the input into a lower dimensional code, the decoder then reconstructs the input only using this code.
+
+### Autoencoder Pipelines
+
+Numalogic provides two types of pipelines for Autoencoders.
+
+#### AutoencoderPipeline
+
+Here we are using `VanillAE`, a Vanilla Autoencoder model.
+
+```python 
+from numalogic.models.autoencoder.variants import Conv1dAE
+from numalogic.models.autoencoder.pipeline import SparseAEPipeline
+
+model = AutoencoderPipeline(
+    model=VanillaAE(signal_len=12, n_features=3), seq_len=seq_len
+)
+model.fit(X_train)
+```
+
+#### SparseAEPipeline
+
+SparseAEPipeline can be used when the training data is sparse. 
```

Comment:
Sparse Autoencoder is not for handling sparse training data. It is to allow overcomplete autoencoders to learn something meaningful about the data, and not just the identity function. Basically to reduce overfitting.

More details here: https://web.stanford.edu/class/cs294a/sparseAutoencoder.pdf You can reference this paper in the doc.